Slack-Apps/week4/homework/responses/4.3-final-project.md

```diff
@@ -0,0 +1,76 @@
+# ✔️Final Project
+
+Congratulations, you are about to complete your final homework assignment! It's time for you to apply the concepts from this week into a project. You will create an app that listens for Github comments and posts in the app's chat whenever this event occurs.
+
+## Setting up Express
+
+You first need to set up Express in order to handle the payload from a GitHub webhook. First, create a folder on your local machine and navigate to the directory in your terminal. Call ```npm install express ``` and wait for the package to download. Next, set up your local slack app in the same directory. 
+
+After the dependencies have been installed, open your app.js file in a text editor and paste the following code:
+
+```javascript
+const { App, ExpressReceiver } = require('@slack/bolt');
+const express = require('express');
+const { config } = require('dotenv');
+
+const app = new App({
+  token: process.env.SLACK_BOT_TOKEN,
+  receiver
+});
+```
+
+This handles all dependencies and objects needed for a basic app.
+
+### Outlining the Post Method
+
+Create an ExpressReceiver object to handle the webhook payload:
+
+```javascript
+const receiver = new ExpressReceiver({ signingSecret: process.env.SLACK_SIGNING_SECRET });
+receiver.router.use(express.json());
+```
+
+This object can be used to invoke the GET and POST functions you read about in a previous issue. 
+
+This project will use a POST function. 
+
+```javascript
+receiver.router.post('/github-comments', async (req, res) => { 
+  const text = ``; 
+  await app.client.chat.postMessage({
+    token: process.env.SLACK_BOT_TOKEN,
+    channel: 'welcome',
+    text,
+  });
+
+  res.sendStatus(200);
+});
+
+```
+
+This is the basic outline of what our app will do. It will listen on a server for a payload, then formulate some sort of message, then post that text in a channel that the bot is in. 
+
+## Setting Up the Webhook
+
+The first parameter in the post method is ```'/github-comments'```, which outlines the endpoint that the webhook payload is received at. This is similar to the ```'/slack/events'``` endpoint that is specified for event subscriptions. 
+
+In the last issue, you learned how to create a webhook. Select or create a GitHub repository and create a new webhook listening for comments. Remember to keep an ngrok server running and **update your slack app** to use the same link for interactivity and to listen for events. Append ```'/github-comments'``` to the ngrok link for the webhook payload URL. 
```

Comment:
I think it would be helpful to directly reference the issues for students to have easy access to tutorials. Since this project is a culmination of all the skills, information from almost all issues are probably applied.

Reply:
@emsesc could you clarify what you mean by a direct reference? Reviewing how to create a webhook was the subject of an entire issue, so I can put a link for that issue. Going over it again seems too redundant.

Reply:
That's exactly what I meant, putting the link for that issue 👍 I agree that going over it again is too redundant.

Reply:
perfect, thank you!